intruder_alarm/src/stack/mod.rs

```diff
@@ -19,32 +20,32 @@ mod tests;
 //-----------------------------------------------------------------------------
 // Public API types
 //-----------------------------------------------------------------------------
-//  List
-/// An intrusive singly-linked list.
+//  Stack
+/// A stack implementation using an intrusive singly-linked list.
 ///
 /// This type is a wrapper around a series of [`Node`]s. It stores [`Link`]s
-/// to the head [`Node`]s and the length of the list.
+/// to the head/top [`Node`]s and the size of the stack (a.k.a len of the list).
 ///
-/// `FromIterator` will return a new list in reverse order because
-/// singly-linked list only provides stack operations.
+/// `FromIterator` will return a new stack (list) in reverse order because this
+/// is a stack.
 ///
 /// # Type parameters
 /// - `T`: the type of the items stored by each `N`
-/// - `N`: the type of nodes in the list
+/// - `N`: the type of nodes in the stack
 /// - `R`: the type of [`OwningRef`] that owns each `N`.
 ///
 /// [`Node`]: trait.Node.html
 /// [`Link`]: ../struct.Link.html
 /// [`OwningRef]: ../trait.OwningRef.html
 #[derive(Default)]
-pub struct List<T, N, R> {
-    /// Link to the head node of the list.
-    head: Link<N>,
+pub struct Stack<T, N, R> {
+    /// Link to the top node of the stack.
+    top: Link<N>,
 
-    /// Length of the list.
-    len: usize,
+    /// Size of the stack.
+    size: usize,
```

Comment:
Since these are private fields, rather than API, I don't really care as much --- the name of this field can be changed whenever we want, to whatever we want, and it won't impact downstream consumers at all. With that said, I think it makes sense to be consistent, but it's not a blocker for merging this PR.

Reply:
I agree. Sorry about the delay in responding. Should I open another ticket for this and doubly renaming?

Reply:
Go for it, or just go ahead and open a PR for it.